packages/gatsby-plugin-image/README.md

```diff
@@ -26,331 +32,225 @@ If you're using the new `GatsbyImage` in addition to `StaticImage`, you'll also
-| image | object          | The image data object returned from a GraphQL query                                                                        |
-| alt   | string          | Passed to the `img` element. Defaults to an empty string                                                                   |
-| sizes | string          | An HTML "sizes" argument, which is passed-though to the image. Can be left blank, when it will be calculated automatically |
-| as    | React Component | The component that wraps the Gatsby Image. Default is `div`                                                                |
-
-# GraphQL resolver
+```js
+// Also OK
 
-We have added a new `gatsbyImageData` resolver to the `ImageSharp` node. Unlike the existing `fixed` and `fluid` resolvers, this returns a
-JSON type, meaning you don't specify the individual fields, but are instead given the whole object. This is because the object is then passed into the `<GatsbyImage>` component. The API is like this:
+// A variable in the same file is fine.
+const width = 300
 
-```graphql
-coverImage: file(relativePath: { eq: "plant.jpg" }) {
-  childImageSharp {
-    gatsbyImageData(width: 720, layout: CONSTRAINED, placeholder: TRACED_SVG)
-  }
+export function Dino()  {
+    // This works because the value can be statically-analyzed
+    const height = width * 16 / 9
+    return <StaticImage src="trex.png" width={width} height={height}>
 }
 ```
 
-You then use the data like this:
-
-```jsx
-import { GatsbyImage, getImage } from "gatsby-plugin-image"
+If you find yourself wishing you could use a prop for the image `src` then it's likely that you should be using a dynamic image.
+
+### Dynamic images
+
+If you need to have dynamic images (such as if they are coming from a CMS), you can load them via GraphQL and display them using the `GatsbyImage` component.
+
+1. **Add the image to your page query.**
+
+   Any GraphQL File object that includes an image will have a `childImageSharp` field that you can use to query the image data. The exact data structure will vary according to your data source, but the syntax is like this:
+
+   ```graphql
+   query {
+     blogPost(id: { eq: $Id }) {
+       title
+       body
+       avatar {
+         childImageSharp {
+           gatsbyImageData(width: 200)
+         }
+       }
+     }
+   }
+   ```
+
+2. **Configure your image.**
+
+   You configure the image by passing arguments to the `gatsbyImageData` resolver. You can change the size and layout, as well as settings such as the type of placeholder used when lazy loading. There are also advanced image processing options available. You can find the full list of options in the API docs.
+
+   ```graphql
+   query {
+     blogPost(id: { eq: $Id }) {
+       title
+       body
+       author
+       avatar {
+         childImageSharp {
+           gatsbyImageData(
+             width: 200
```

Comment:
```suggestion
             width: 200,
```

Reply:
Commas are optional when it's multiline. Prettier removes them.

Reply:
I legit just realized this when I was doing it on my site. Strange.

Reply:
I accepted all of the suggesitons, but then started to edit locally before committing then realised prettier just removed them again!

Reply:
Ya, I think just leave them out. Hopefully people won't have copy/paste issues. But if it's valid syntax they shouldn't.